foyer/forcefields/oplsaa.xml

```diff
@@ -516,23 +516,23 @@
   <Type name="opls_516" class="opls_516" element="C" mass="12.011"/>
   <Type name="opls_517" class="opls_517" element="C" mass="12.011"/>
   <Type name="opls_518" class="opls_518" element="C" mass="12.011"/>
-  <Type name="opls_520" class="opls_520" element="N" mass="14.0067"/>
-  <Type name="opls_521" class="opls_521" element="C" mass="12.011"/>
-  <Type name="opls_522" class="opls_522" element="C" mass="12.011"/>
-  <Type name="opls_523" class="opls_523" element="C" mass="12.011"/>
-  <Type name="opls_524" class="opls_524" element="H" mass="1.008"/>
-  <Type name="opls_525" class="opls_525" element="H" mass="1.008"/>
-  <Type name="opls_526" class="opls_526" element="H" mass="1.008"/>
+  <Type name="opls_520" class="opls_520" element="N" mass="14.0067" def="[N;X2;r6]1[C;X3;r6][C;X3;r6][C;X3;r6][C;X3;r6][C;X3;r6]1" desc="N in pyridine"/>
+  <Type name="opls_521" class="opls_521" element="C" mass="12.011" def="[C;X3;r6]([N;%opls_520])" desc="C1 in pyridine"/>
+  <Type name="opls_522" class="opls_522" element="C" mass="12.011" def="[C;X3;r6]([C;%opls_521])" desc="C2 in pyridine" overrides="opls_142"/>
+  <Type name="opls_523" class="opls_523" element="C" mass="12.011" def="[C;X3;r6]([C;%opls_522])[C;%opls_522]" desc="C3 in pyridine" overrides="opls_142"/>
+  <Type name="opls_524" class="opls_524" element="H" mass="1.008" def="H[C;%opls_521]" desc="H1 in pyridine" overrides="opls_144"/>
+  <Type name="opls_525" class="opls_525" element="H" mass="1.008" def="H[C;%opls_522]" desc="H2 in pyridine" overrides="opls_144"/>
+  <Type name="opls_526" class="opls_526" element="H" mass="1.008" def="H[C;%opls_523]" desc="H3 in pyridine" overrides="opls_144"/>
   <Type name="opls_527" class="opls_527" element="N" mass="14.0067"/>
   <Type name="opls_528" class="opls_528" element="C" mass="12.011"/>
   <Type name="opls_529" class="opls_529" element="H" mass="1.008"/>
-  <Type name="opls_530" class="opls_530" element="N" mass="14.0067"/>
-  <Type name="opls_531" class="opls_531" element="C" mass="12.011"/>
-  <Type name="opls_532" class="opls_532" element="C" mass="12.011"/>
-  <Type name="opls_533" class="opls_533" element="C" mass="12.011"/>
-  <Type name="opls_534" class="opls_534" element="H" mass="1.008"/>
-  <Type name="opls_535" class="opls_535" element="H" mass="1.008"/>
-  <Type name="opls_536" class="opls_536" element="H" mass="1.008"/>
+  <Type name="opls_530" class="opls_530" element="N" mass="14.0067" def="[N;X2;r6]1[C;X3;r6][C;X3;r6][C;X3;r6][N;X2;r6][C;X3;r6]1" desc="N in pyrimidine"/>
+  <Type name="opls_531" class="opls_531" element="C" mass="12.011" def="[C;X3;r6]([N;%opls_530])[N;%opls_530]" desc="C2 in pyrimidine"/>
+  <Type name="opls_532" class="opls_532" element="C" mass="12.011" def="[C;X3;r6]([N;%opls_530])C" desc="C4 in pyrimidine"/>
```

Comment:
Might want to define the second carbon as explicitly part of the ring, i.e. `[C;X3;r6]([N;%opls_530])[C;X3;r6]`